### DISCUSSION TITLE
Damage Tracking (probably) broken on (proprietary) NVidia, causing huge "input lag"

### DISCUSSION THREAD

**Discussion by shinobiultra** [hyprwm/Hyprland#4575] [Q&A]
Hi,

so I'm running Arch & Hyprland on:

`
Graphics:
  Device-1: Intel Raptor Lake-S GT1 [UHD Graphics 770] driver: i915 v: kernel
  Device-2: NVIDIA TU106 [GeForce RTX 2060 SUPER] driver: nvidia
    v: 545.29.06
  Display: wayland server: X.org v: 1.21.1.11 with: Xwayland v: 23.2.4
    compositor: Hyprland driver: X: loaded: modesetting,nvidia dri: iris
    gpu: i915,nvidia,nvidia-nvswitch resolution: 1: 1680x1050~60Hz
    2: 3440x1440~100Hz 3: 1920x1080~60Hz
  API: EGL Message: EGL data requires eglinfo. Check --recommends.
`

Now, I've got two monitors (60Hz, 1050p and 1080p) connected to my iGPU and one (3440x1440@144) connected to NVidia. I am running `nvidia-open-dkms` / `nvidia-dkms` (didn't change the problem) as I rely on CUDA.

I've done everything as per the wiki but I always ended up with 30Hz main monitor. That was until I found this discussion https://github.com/hyprwm/Hyprland/discussions/4240 and it (setting _nvidia_anti_flicker_ to false) fixed everything! Well, more like it changed the issue to another one.

Now, I've had insane "input lag", like I'd type in terminal and it would take 0.1 - 0.5s for the character to appear. And deleting was insanely slow and if I just pressed down backspace, it would delete the characters slowly but keep the first character and it would only appear deleted once I *released* the backspace!

And stuff like this drove me crazy but made me suspect there's just some damage tracking problem, so I've quickly found debug option to just turn it off and even tho the [wiki](https://wiki.hyprland.org/Configuring/Variables/#debug) clearly says 

> Do **not** change.

I changed it. And it fixed everything. 

I.e., doing:

`
debug {
    damage_tracking =  0
}
`

fixed it. Setting it to 1 or (the default) 2 keeps the problem unchanged.

My question is - what ? Why ? Can we fix this in a better way ? Can we at least document this possible workaround for any other lost soul enduring this hell with proprietary driver(s) & Wayland ?

Any technical insight, help, workaround or any other (re)solution greatly welcome.

I will thus continue walking among you guys, adding "I use hyprland btw" to my pickup line but know, that I did what our master wiki prohibits...

Cheers

**Comment by vaxerski** [comment]
have you tried disabling nvidia_anti_flicker instead

**Reply by shinobiultra** [reply]
Yes, without it I was stuck at 30Hz. With it I was able to get any refresh rate I chose but at the cost of this huge input lag. I'll try to clarify this in the post.

Cheers

**Reply by Vilen23** [reply]
Did you find any fix?
